jina/drivers/predict.py

```diff
@@ -0,0 +1,163 @@
+from typing import Iterable, List, Any, Union
+
+import numpy as np
+
+from . import BaseExecutableDriver
+from .helper import extract_docs
+from ..proto.ndarray.generic import GenericNdArray
+
+if False:
+    from ..proto import jina_pb2
+
+
+class BasePredictDriver(BaseExecutableDriver):
+    """Drivers inherited from this Driver will bind :meth:`predict` by default """
+
+    def __init__(self, executor: str = None, method: str = 'predict', *args, **kwargs):
+        super().__init__(executor, method, *args, **kwargs)
+
+
+class BaseLabelPredictDriver(BasePredictDriver):
+
+    def __init__(self, output_tag: str = 'prediction', *args, **kwargs):
+        """
+
+        :param output_tag: output label will be written to ``doc.tags``
+        :param args:
+        :param kwargs:
+        """
+        super().__init__(*args, **kwargs)
+        self.output_tag = output_tag
+
+    def _apply_all(
+            self,
+            docs: Iterable['jina_pb2.Document'],
```

Comment:
i think `Sequence` is a better type hints since we expect the interable object to be ordered

Reply:
Then this should be changed globally in a separate PR

Reply:
raised #1196